### PR TITLE
Add by filter deletion

### PR DESCRIPTION
**Pull request by antoinebhs**

### CODE REVIEW

File: src/components/dialogs/network-modifications/by-filter-deletion/by-filter-deletion-dialog.js
Comment (flomillot) [inline src/components/dialogs/network-modifications/by-filter-deletion/by-filter-deletion-dialog.js:1]: For new files we try to do TypeScript if I'm right, not sure if it's mandatory yet

File: src/components/dialogs/network-modifications/by-filter-deletion/by-filter-deletion-form.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        return Object.values(EQUIPMENT_TYPES).filter(
+            (equipmentType) => !equipmentTypesToExclude.has(equipmentType)
+        );
+    }, []);
+
+    const handleEquipmentTypeChange = useCallback(() => {
+        setValue(FILTERS, []);
+    }, [setValue]);
+
+    const filtersField = useMemo(() => {
```
Comment (flomillot) [inline src/components/dialogs/network-modifications/by-filter-deletion/by-filter-deletion-form.tsx:52]: Why `useMemo` only on this one ?

File: src/components/dialogs/network-modifications/by-filter-deletion/by-filter-deletion-form.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            EQUIPMENT_TYPES.HVDC_CONVERTER_STATION,
+            EQUIPMENT_TYPES.BUS,
+            EQUIPMENT_TYPES.HVDC_LINE, // HVDC lines are not supported because of specific update related to HVDC LCC deletion.
+        ]);
+        return Object.values(EQUIPMENT_TYPES).filter(
+            (equipmentType) => !equipmentTypesToExclude.has(equipmentType)
+        );
+    }, []);
+
+    const handleEquipmentTypeChange = useCallback(() => {
```
Comment (SlimaneAmar) [inline src/components/dialogs/network-modifications/by-filter-deletion/by-filter-deletion-form.tsx:51]: Need a `useCallback` ?

File: src/services/study/network-modifications.js [hunk 3]
```diff
@@ -1555,6 +1555,38 @@ export function deleteEquipment(
     });
 }
 
+export function byFilterDeleteEquipment(
```
Comment (SlimaneAmar) [inline src/services/study/network-modifications.js:1558]: Rename by `deleteEquipmentByFilter` ?

File: src/components/dialogs/network-modifications/by-filter-deletion/by-filter-deletion-dialog.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+                snackError({
+                    messageTxt: error.message,
+                    headerId: 'UnableToDeleteEquipment',
+                });
+            });
+        },
+        [currentNodeUuid, editData, snackError, studyUuid]
+    );
+
+    const clear = useCallback(() => {
```
Comment (SlimaneAmar) [inline src/components/dialogs/network-modifications/by-filter-deletion/by-filter-deletion-dialog.js:116]: `useCallback` ?

File: src/components/dialogs/network-modifications/by-filter-deletion/by-filter-deletion-dialog.js [hunk 5]
```diff
@@ -0,0 +1,157 @@
+/**
```
Comment (SlimaneAmar) [inline src/components/dialogs/network-modifications/by-filter-deletion/by-filter-deletion-dialog.js:1]: Why not TS ?
For new dialog !

File: src/components/dialogs/network-modifications/by-filter-deletion/by-filter-deletion-form.tsx (hunk 1 above)
Comment (antoinebhs) [reply src/components/dialogs/network-modifications/by-filter-deletion/by-filter-deletion-form.tsx:55]: Added

File: src/components/dialogs/network-modifications/by-filter-deletion/by-filter-deletion-dialog.js
Comment (SlimaneAmar) [reply src/components/dialogs/network-modifications/by-filter-deletion/by-filter-deletion-dialog.js:1]: Yes it's mandatory !

Comment (antoinebhs) [reply src/components/dialogs/network-modifications/by-filter-deletion/by-filter-deletion-dialog.js:1]: @SlimaneAmar @flomillot, I took example on the recent implementations of by formula and tabular modification, I translated the file to TS :+1:

**Review by flomillot — APPROVED** [review]
Code OK

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**